UXIT-Contractors/UXIT-Planner-App
# Feat aanpassen rooster

## Description
Feature that allows admins to view a table of all planned shifts. These are sorted by date and time, and display the current staffed users.

Shifts can be expanded, displaying more details and allowing the admin to change the properties of the shift, such as:
- start time
- end time
- required amount of volunteers for the generate schedule to assign
- delete shift

In addition to this, the admin can manually change staffing assigned to a shift by doing the following:
- remove staffing
- select a volunteer/employee to add as staffing

## What changed?
Added Role.EMPLOYEE in role.ts and made a duo of mock employees in seed.ts

## How to test:
1. Navigate to the branch feat_AanpassenRooster and pull.
2. Ensure you can log in as admin by looking at your env file.
3. Run it with "npm run seed" followed up by "npm run dev".
4. Go to the admin panel and press "Handmatige aanpassingen".
5. You will see a blank table, these are shifts where no one has been assigned as staffing.
6. Go back to the admin panel and generate roster for the current month.
7. Back again to the shift overview and now you'll see all staffed users.
8. Click on a row and it will expand.
9. Now you can do the following:
- Press the button on the top left to minimize the shift
- Press the button the top right to delete the shift
- Change the Starttijd field and press on the button next to it
- Change the Eindtijd field and press on the button next to it
- Click on the + / - buttons under Benodigde vrijwilligers and press the button next to it
- Press the trash can button next to an assigned staffing to remove it
- Select a shift type at the selector at the bottom of the shift
- Select a volunteer or employee, based on the selected shift type (if a user is staffed already, you won't see them in the dropdown so you can't add them twice, if you remove a staffing you'll see you can select that user again in the dropdown.)
- Press the button below the select field to add the staffing to the shift
10. Remember your changes and refresh the page.
11. If you expand the same shift again you'll see everything is still there 👍

## Some screenshots if you're lazy:
Overview shifts:
![image](https://github.com/UXIT-Contractors/UXIT-Planner-App/assets/113612336/c6568548-1cb3-41f2-9a4e-80e53f28d32f)

Expanded shift:
![image](https://github.com/UXIT-Contractors/UXIT-Planner-App/assets/113612336/89b24d73-c3eb-4a85-ab5a-fe60e374e18a)

Changed time:
![image](https://github.com/UXIT-Contractors/UXIT-Planner-App/assets/113612336/a7a945ab-4aa7-4724-b466-ff5ce9ebcaf0)

Changed staffing required:
![image](https://github.com/UXIT-Contractors/UXIT-Planner-App/assets/113612336/25e869bf-7b45-4c0b-9e0e-b23ba795a096)

Feedback of properties changed:
![image](https://github.com/UXIT-Contractors/UXIT-Planner-App/assets/113612336/93a01787-7949-4377-8951-3693d2e47230)

Adding a volunteer:
![image](https://github.com/UXIT-Contractors/UXIT-Planner-App/assets/113612336/b558fc3b-8e31-4f78-98a3-cff5e22d5e7d)

Volunteer added:
![image](https://github.com/UXIT-Contractors/UXIT-Planner-App/assets/113612336/408e0f45-3486-4b67-a194-f9b54d95e355)

New shift overview with updated properties:
![image](https://github.com/UXIT-Contractors/UXIT-Planner-App/assets/113612336/ccdf8cdc-26c0-496a-b6cd-19ab8e051c05)
